src/awesome/tag.rs

```diff
@@ -132,7 +135,30 @@ fn set_activated<'lua>(lua: &'lua Lua, (obj, val): (AnyUserData<'lua>, bool))
         }
         tag.activated = val;
     }
-    if !val {
+    let activated_tags = lua.named_registry_value::<Table>(TAG_LIST)?;
+    let activated_tags_count = activated_tags.len()?;
+    if val {
+        let index = activated_tags_count + 1;
+        activated_tags.set(index, obj.clone())?;
+    } else {
+        // Find and remove the tag in/from the list of tags
+        {
+            let tag_ref = &*obj.borrow::<TagState>()? as *const _;
+            let mut found = false;
+            for pair in activated_tags.clone().pairs::<Integer, AnyUserData>() {
+                let (key, value) = pair?;
+                if tag_ref == &*value.borrow::<TagState>()? as *const _ {
+                    found = true;
+                    // Now remove this by shifting everything down...
+                    for index in key .. activated_tags_count-1 {
+                        activated_tags.set(index,
+                                           activated_tags.get::<_, Value>(index + 1)?)?;
+                    }
+                    break;
+                }
+            }
+            assert!(found);
```

Comment:
Do we actually want to crash the Lua thread if the tag isn't found?

Reply:
Dunno. This assertion should always hold, so I'd say that it is a "correct" use of `assert!` (put differently: "This assert can never fail!" (famous last words)). I don't know if something like `error!` makes more sense here. Or perhaps a `debug_assert!`?

Reply:
Ok, I'm fine with failing early here, it's better than letting a bug slip through.

Reply:
I'd also rather have the tags as keys in the table so that I do not have to mess with an array like this... but awesome does not do that either (and "cheats" by using `memmove` to remove an entry, which is a lot cheapter than what has to be done here).